tasks/eva_2124/correct_assembly_for_study.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+import hashlib
+from argparse import ArgumentParser
+
+import pymongo
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+from tasks.eva_2124.load_synonyms import load_synonyms_for_assembly
+
+def get_SHA1(variant_rec):
+    """Calculate the SHA1 digest from the ref, study, contig, start, ref, and alt attributes of the variant"""
+    h = hashlib.sha1()
+    keys = ['seq', 'study', 'contig', 'start', 'ref', 'alt']
+    h.update('_'.join([str(variant_rec[key]) for key in keys]).encode())
+    return h.hexdigest().upper()
+
+
+def get_genbank(synonym_dictionaries, contig):
+    """
+    returns a tuple (genbank, was_already_genbank) or raises an exception if the contig was not found
+    """
+    by_name, by_assigned_molecule, by_genbank, by_refseq, by_ucsc = synonym_dictionaries
+    if contig in by_name:
+        return by_name[contig]['genbank'], False
+
+    if contig in by_assigned_molecule:
+        return by_assigned_molecule[contig]['genbank'], False
+
+    if contig in by_ucsc:
+        return by_ucsc[contig]['genbank'], False
+
+    if contig in by_refseq and by_refseq[contig]['is_genbank_refseq_identical']:
+        return by_refseq[contig]['genbank'], False
+
+    if contig in by_genbank:
+        return contig, True
+
+    raise Exception('could not find synonym for contig {}'.format(contig))
```

Comment:
What is the option if this occurs. 
We will have to change the script so it can accept an assembly report from the command line ?